### PR TITLE
LP-1592 LP-1593 Unit test for badge assignment, my badges and trophy case

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
Unit test for badge assignment, my badges and trophy case

### CODE REVIEW

File: openedx/features/badging/tests/test_helpers.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        badge1 = BadgeFactory(type=TEAM_PLAYER[TEAM_PLAYER_ENTRY_INDEX], threshold=2)
+        badge2 = BadgeFactory(type=self.type_conversationalist, threshold=2)
+        badge3 = BadgeFactory(type=self.type_conversationalist, threshold=5)
+
+        badges = badge_helpers.get_course_badges(self.user, self.course1.id, earned_badges=mock.ANY)
+
+        expected_result = {
+            'badges': {
+                'conversationalist': [{
+                    'description': None, 'image': badge2.image, 'threshold': 2L,
```
Comment (saad-waqar) [inline openedx/features/badging/tests/test_helpers.py]: what is this L?

Comment (zahid-arbisoft) [reply openedx/features/badging/tests/test_helpers.py]: **L** is long integer literal. It makes integer [long integer](https://docs.python.org/2.0/ref/integers.html). Though it is not necessary to make threshold long integer here, integer is just fine but it doesn't break anything

File: openedx/features/badging/tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+        course_badges = list(
+            Badge.objects.all().values()
+        )
+        earned_badges = [user_badge_1, user_badge_2]
+        badge_helpers.add_badge_earned_date(self.course1.id, course_badges, earned_badges)
+
+        # get badges by index, since list maintains insertion order
+        self.assertIsNotNone(course_badges[0]['date_earned'])
+        self.assertRaises(KeyError, lambda: course_badges[1]['date_earned'])
```
Comment (emraan004) [inline openedx/features/badging/tests/test_helpers.py]: let's verify the result of this test

File: openedx/features/badging/tests/test_helpers.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        ])
+
+        self.assertEqual(expected_return_value, trophycase_dict)
+
+        # assert order of courses by display name
+        course_detail = trophycase_dict.values()
+        self.assertEqual(expected_return_value['test1/123/1'], course_detail[0])
+        self.assertEqual(expected_return_value['test2/123/1'], course_detail[1])
+
+    @mock.patch('openedx.features.badging.helpers.Badge.objects.all')
```
[the remark this answers is not in the document]
Comment (zahid-arbisoft) [reply openedx/features/badging/tests/test_helpers.py]: I have updated test method name

File: openedx/features/badging/tests/test_helpers.py (hunk 2 above)
Comment (zahid-arbisoft) [reply openedx/features/badging/tests/test_helpers.py]: It is passing, I have also added documentation

**Pull request merged**